### PR TITLE
Fix termination bug

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: module-rt/tests/multiple.rs [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            let handles_1_to_2 = port1.export(&[if j > i {
+                j - 1
+            } else {
+                j
+            }]);
+            let handles_2_to_1 = port2.export(&[if i > j {
+                i - 1
+            } else {
+                i
+            }]);
```
Comment (majecty) [inline module-rt/tests/multiple.rs]: Could you tell me why does this test need `i - 1`?

File: module-rt/tests/multiple.rs [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        let pizzas = pool.split_off(n);
+        ServiceRef::create_export(Box::new(SimplePizzaBox {
+            _pizzas: pizzas,
+        }) as Box<dyn PizzaBox>)
+    }
+}
+struct ModuleA {
+    pizza_stores: Vec<Box<dyn PizzaStore>>,
+    pizza_pool: Arc<RwLock<Vec<Box<dyn Pizza>>>>,
+    pizza_boxes: Vec<Box<dyn PizzaBox>>,
```
Comment (majecty) [inline module-rt/tests/multiple.rs:82]: Could you add a comment that what this test wants to do?

File: module-rt/tests/multiple.rs [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+        let pizza_box_n = rng.gen_range(10, 100);
+        let mut pizza_boxes: Vec<Box<dyn PizzaBox>> = Vec::new();
+        for _ in 0..pizza_box_n {
+            pizza_boxes.push(self.pizza_stores.choose(&mut rng).unwrap().wrap_pizza_box().unwrap_import().into_proxy())
+        }
+        self.pizza_boxes = pizza_boxes;
+
+        Vec::new()
+    }
```
Comment (majecty) [inline module-rt/tests/multiple.rs]: It's hard to understand what is happening in this test.
Please write comments that explain the scenario.

File: module-rt/src/bootstrap.rs [hunk 4]
```diff
@@ -85,6 +89,10 @@ impl<T: UserModule + 'static> FoundryModule for ModuleContext<T> {
         ServiceRef::create_export(port_ as Arc<RwLock<dyn Port>>)
     }
 
+    fn seal(&mut self) {
```
Comment (majecty) [inline module-rt/src/bootstrap.rs]: I couldn't understand the concept of `seal` here.
Could you explain it to me?

**Review by majecty — CHANGES_REQUESTED** [review]

File: module-rt/tests/multiple.rs (hunk 1 above)
Comment (junha1) [reply module-rt/tests/multiple.rs]: Please check the comment

File: module-rt/tests/multiple.rs (hunk 2 above)
Comment (junha1) [reply module-rt/tests/multiple.rs:82]: updated

File: module-rt/tests/multiple.rs (hunk 3 above)
Comment (junha1) [reply module-rt/tests/multiple.rs]: updated

File: module-rt/src/bootstrap.rs (hunk 4 above)
Comment (junha1) [reply module-rt/src/bootstrap.rs]: https://github.com/CodeChain-io/foundry/commit/8efc4fc83d44e6e5fff95a1b67b0c38afc341e5c
It notifies the module that the bootstrapping is done

File: module-rt/tests/multiple.rs [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+        // This will prepare its own pizza pool and gather pizza boxes.
+        // Note that other instances' prepared pizza pools are for my pizza boxes,
+        // but my pizza pool is for the other module's pizza boxes.
+
+        for _ in 0..pizza_n {
+            // Add a pizza from a randomly chosen pizza store
+            self.pizza_pool
+                .write()
+                .push(self.pizza_stores.choose(&mut rng).unwrap().create_pizza().unwrap_import().into_proxy());
```
Comment (majecty) [inline module-rt/tests/multiple.rs]: Could you split this line `self.pizza_stores.choose(&mut rng).unwrap().create_pizza().unwrap_import().into_proxy()` into multiple variables?
It's hard to follow the whole line.

File: module-rt/tests/multiple.rs [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        }
+
+        let pizza_box_n = rng.gen_range(10, 100);
+        let mut pizza_boxes: Vec<Box<dyn PizzaBox>> = Vec::new();
+        for _ in 0..pizza_box_n {
+            // Add a pizza box from a randomly chosen pizza store
+            pizza_boxes.push(self.pizza_stores.choose(&mut rng).unwrap().wrap_pizza_box().unwrap_import().into_proxy())
```
Comment (majecty) [inline module-rt/tests/multiple.rs]: Could you split the `self.pizza_stores.choose(&mut rng).unwrap().wrap_pizza_box().unwrap_import().into_proxy()` line? It is hard to follow.

**Review by majecty — CHANGES_REQUESTED** [review]
@junha1 Do you know where the concept `seal` came from?
I want to know which metaphor does it have

File: module-rt/tests/multiple.rs (hunk 5 above)
Comment (junha1) [reply module-rt/tests/multiple.rs]: @majecty Would you suggest the code? where to cut and what the name of variable would be?

File: module-rt/src/bootstrap.rs (hunk 4 above)
Comment (majecty) [reply module-rt/src/bootstrap.rs]: `seal` in Linker context has a meaning. It will make Linker or Linkable not to link anymore.
However `seal` in module context is ambiguous. Please write a comment over the `seal` function or change the name.

File: module-rt/tests/multiple.rs (hunk 5 above)
Comment (majecty) [reply module-rt/tests/multiple.rs]:
```rust
            let random_pizza_store = self.pizza_stores.choose(&mut rng).unwrap();
            let imported_pizza_service = random_pizza_store.create_pizza().unwrap_import();
            let pizza_proxy = imported_pizza_service.into_proxy();

            // Add a pizza from a randomly chosen pizza store
            self.pizza_pool.write().push(pizza_proxy);
```

File: module-rt/tests/multiple.rs (hunk 6 above)
Comment (majecty) [reply module-rt/tests/multiple.rs]:
```rust
            let random_pizza_store = self.pizza_stores.choose(&mut rng).unwrap();
            let imported_pizza_box_service = random_pizza_store.wrap_pizza_box().unwrap_import();
            let pizza_box_proxy = imported_pizza_box_service.into_proxy();

            // Add a pizza box from a randomly chosen pizza store
            pizza_boxes.push(pizza_box_proxy)

```

**Review by majecty — APPROVED** [review]
